unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -249,20 +252,42 @@ loadReferentType codebase = \case
           ++ show r
 
 data TermEntry v a = TermEntry
-  { termEntryReferent :: Referent,
-    termEntryName :: HQ'.HQSegment,
```

Comment:
I don't like having `HashQualified`'s in the the API boundary, since it's really unclear whether they're expected to have a hash or not. Instead I return as much information as possible, and the decision can be made explicitly later on.

Reply:
Yeah